rtexif/rtexif.cc

```diff
@@ -3322,11 +3315,20 @@ int ExifManager::createJPEGMarker (const TagDirectory* root, const rtengine::pro
     }
 
     cl->sort ();
-    int size = cl->write (8, buffer + 6);
+    bufferSize = cl->calculateSize() + 8 + 6;
+    buffer = new unsigned char[bufferSize]; // this has to be deleted in caller
+    memcpy (buffer, "Exif\0\0", 6);
+    sset2 ((unsigned short)order, buffer + offs, order);
+    offs += 2;
+    sset2 (42, buffer + offs, order);
+    offs += 2;
+    sset4 (8, buffer + offs, order);
+
+    int endOffs = cl->write (8, buffer + 6);
 
     delete cl;
 
-    return size + 6;
+    return endOffs;
```

Comment:
Why does this not have `+ 6` while the old code does? Or maybe this should be removed because the return value is not used anywhere anymore.

Reply:
> Why does this not have + 6 while the old code does?
It's my bug, correct value is `endOffs + 6`... thanks for pointing out.

> maybe this should be removed because the return value is not used anywhere anymore

I agree. At the time in wrote this code, I was trying to keep the changes to a minimum. 

I think to make `ExifManager::createJPEGMarker` does not return a value, in next commit.